Fix OperatorCatalog hash.

mantomas opened the pull request.

mantomas committed:
Fix OperatorCatalog hash.

mporrato reviewed (changes requested):
Looks like a copy-paste mistake. Can you also fix the use of `operator_name` in `__eq__()` and `__lt__()` and perhaps a test case to prevent regressions, please?

mantomas committed:
Fix comparison

mantomas commented:
Changed and updated the test so there are two catalogs with the same operator instead of two different operators in the same catalog. This should also cover the regression.
